src/state_initialization.jl

```diff
@@ -92,17 +92,20 @@ function condition_on!(x::SRGaussian, H::AbstractMatrix, data::AbstractVector,
                        meascache, Kcache, Kcache2, covcache, Mcache)
     z, S = meascache
 
-    mul!(z, H, x.μ)
+    _matmul!(z, H, x.μ)
     X_A_Xt!(S, x.Σ, H)
+    @assert isdiag(S)
+    S = Diagonal(S)
 
-    mul!(Kcache, x.Σ, H')
-    K = mul!(Kcache2, Kcache, inv(S))
+    _matmul!(Kcache, x.Σ.mat, H')
+    Kcache2 .= Kcache ./ S.diag'
+    K = Kcache2
 
-    mul!(x.μ, K, data - z, 1, 1)
+    _matmul!(x.μ, K, data - z, 1, 1)
     # x.μ .+= K*(data - z)
 
     D = length(x.μ)
-    mul!(Mcache, K, H, -1, 0)
+    _matmul!(Mcache, K, H, -1, 0)
     @inbounds @simd for i in 1:D
```

Comment:
```suggestion
    @inbounds @simd ivdep for i in 1:D
```

if not aliasing.

Reply:
Thanks for the comment! I have to admit, I'm not sure what's happening here. Could you give me a brief explanation or point me to somewhere? In any case, I'll add that change here and into one other loop.

Reply:
`ivdep` is telling it that, even if it cannot prove independence (which it almost certainly cannot at the level you're working at), then some SIMD optimizations should still be turned on. So you should only use it if you know the arrays in your loop are all different arrays, but it will optimize them a bit more in that case.